### ISSUE TITLE
Default admin password

### ISSUE THREAD

**Issue by brinycoolie** [greenpau/caddy-security#63]
Hi,
Once the plugin is installed and the  webadmin user ID is entered what is the correct password?
 I am using version 1.0.17 of the plugin.
Caddy is v2.4.6 h1:HGkGICFGvyrodcqOOclHKfvJC0qTU7vny/7FhYp9hNw=
I am starting caddy with user caddy:caddy by use of systemd and writing the initial users.json to /var/lib/caddy successfully. Thus;
{
  "version": "1.0.17",
  "policy": {
    "password": {
      "keep_versions": 10,
      "min_length": 8,
      "max_length": 128,
      "require_uppercase": false,
      "require_lowercase": false,
      "require_number": false,
      "require_non_alpha_numeric": false,
      "block_reuse": false,
      "block_password_change": false
    },
    "user": {
      "min_length": 3,
      "max_length": 50,
      "allow_non_alpha_numeric": false,
      "allow_uppercase": false
    }
  },
  "revision": 2,
  "last_modified": "2022-02-28T17:05:47.02932396Z",
  "users": [
    {
      "id": "2e6c223e-ef2b-4036-bcc6-8f562ce21692",
      "username": "webadmin",
      "email_address": {
        "address": "webadmin@localdomain.local",
        "domain": "localdomain.local"
      },
      "email_addresses": [
        {
          "address": "webadmin@localdomain.local",
          "domain": "localdomain.local"
        }
      ],
      "passwords": [
        {
          "purpose": "generic",
          "algorithm": "bcrypt",
          "hash": "$2a$10$VvjXt5P.tEh25zJpvlL/fePoJ5aChHk389eDXqkMZaDz/1p6DyPL2",
          "cost": 10,
          "expired_at": "0001-01-01T00:00:00Z",
          "created_at": "2022-02-28T17:05:46.942999899Z",
          "disabled_at": "0001-01-01T00:00:00Z"
        }
      ],
      "created": "2022-02-28T17:05:46.942998945Z",
      "last_modified": "2022-02-28T17:05:47.0293181Z",
      "roles": [
        {
          "name": "admin",
          "organization": "authp"
        }
      ]
    }
  ]
}

without me being able to initially login as webadmin to make config cahnges I think I am stuck.
Thanks in advance for some pointers.

brian

**Comment by greenpau** [comment]
@brinycoolie , remove `users.json` and restart the server. After the restart, you will see the log entry that would have `webadmin@localdomain.local` and the password for the `webadmin` user. It is a random password. After that, login and change the password.

Alternatively, see https://authp.github.io/docs/authenticate/local/password-management to change the hash you have, i.e. `$2a$10$VvjXt5P.tEh25zJpvlL/fePoJ5aChHk389eDXqkMZaDz/1p6DyPL2` to something that you would generate with `bcrypt-cli`.

**Comment by brinycoolie** [comment]
Thanks Paul,
I did not see that the hash was randomly generated each creation, so I assumed it was hardcoded.
Also I had quite a hard time getting my tired old head around the new paradigm with Caddy log files so I could even see the  new random hash. My bad! Got there in the end.
 I wanted to avoid, if possible deleting the entire file since it would also delete any users there so I wanted to work on the second option. I found that the go bitnami implementation would not install on my system.
see:
go version go1.13.8 linux/amd64

brian@beetle-1:~$ go get -u github.com/bitnami/bcrypt-cli

then, bcrypt-cli --help
bcrypt-cli: command not found

brian@beetle-1:~$ bcrypt

Command 'bcrypt' not found, did you mean:

  command 'ccrypt' from deb ccrypt (1.11-1)
  command 'crypt' from deb mcrypt (2.6.8-4)
  command 'zcrypt' from deb barnowl (1.10-1build1)
  command 'mcrypt' from deb mcrypt (2.6.8-4)
  command 'scrypt' from deb scrypt (1.3.0-3)

Try: sudo apt install <deb name>

So I experimented with several javascript based bcrypt command line tools and they did not crosscheck with the go version implemented in the security plugin, so I continued to experiment and tried the htpasswd from apache-utils and it seems to crosscheck and roundtrip well with the security plugin. Even to (at least) two cost functions 10 and 16. however 18 did not finish in a timely manner.
Now I have a bit of flexibility with the users file.

Thanks again.

brian
